File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    vector<int> dp;
    int help(int i){
        if(i<=2) return i;
        if(dp[i]!=-1) return dp[i];
        return dp[i] = help(i-1) + help(i-2);
    }
    int climbStairs(int n) {
    dp.resize(n+1,-1);
    return help(n);    
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    int[] dp = new int[46];
    int help(int n){
        if(n==1) return 1;
        if(n==2) return 2;
        if(dp[n]!=0) return dp[n];
        return dp[n] = help(n-1) + help(n-2); 
    }
    public int climbStairs(int n) {
    return help(n);  
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna